### PR TITLE
book-store: Implement exercise

### PR DESCRIPTION
**Pull request by jgr**
## Why?
Ruby Exercism was missing this exercise.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or enhancement that would cause existing functionality to change)
- [ ] Gardening (code and/or documentation cleanup)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] My change relies on a pending issue/pull request
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by Insti** [comment]
Thanks @jgr let us know if you need any help.

Out of interest: Which documentation have you referred to for making this PR?

**Comment by jgr** [comment]
Howdy @Insti! I'm following [this guide to adding an exercism exercise to another language](https://github.com/exercism/docs/blob/master/you-can-help/implement-an-exercise-from-specification.md). I regret that I've lost the context of how I ended up on that document! Is it out of date?

**Comment by Insti** [comment]
Thanks. That is the most up to date document.
I want to [remove the section about creating an empty WIP pull request.](https://github.com/exercism/docs/pull/108) so I was interested if that was the document you'd gotten the idea from.

**Comment by Insti** [comment]
Have you made any progress on this @jgr ?

**Comment by jgr** [comment]
@Insti yeah, sorry I haven't pushed anything up! I'm just wrapping my head around the generator setup now, I'll let you know if I have any questions.

**Comment by Insti** [comment]
Even partial commits are helpful, so we can tell you you're on the right track..

We should probably write a guide to creating new exercises and generators.

**Comment by jgr** [comment]
@Insti I'd love a review of this PR now, I think I have all the bits required of a new exercise.  Here's a list of the files in the PR and how they were created, I look forward to your feedback!

- `config.json` I added a section to this file, per [this document](https://github.com/exercism/docs/blob/master/you-can-help/implement-an-exercise-from-specification.md#configuring-the-exercise).
- `exercises/book-store/.meta/generator/book_store_case.rb` I wrote this, following the directions [here](https://github.com/exercism/ruby#implementing-a-generator).
- `exercises/book-store/.meta/solutions/.version` I added this manually, because I saw it referenced in the README a few places.  It isn't clear to me how it is used, or if I was supposed to add it manually or expect it to be generated somehow.
- `exercises/book-store/.meta/solutions/book_store.rb` This is my sample solution.
- `exercises/book-store/README.md` I generated this using the Configlet tool.  It wasn't clear to me if I was supposed to do this myself or not.
- `exercises/book-store/book_store_test.rb` I generated this using `./bin/generate book-store`.

**Comment by Insti** [comment]
Great, thanks. I'm a bit busy a t the moment, I'll check this out later in the week.

File: exercises/book-store/book_store_test.rb [hunk 1]
```diff
@@ -0,0 +1,92 @@
+require 'minitest/autorun'
+require_relative 'book_store'
+
+# Common test data version: 1.0.1 087ad69
```
Comment (Insti) [inline exercises/book-store/book_store_test.rb]: There's a new version of the canonical-data.
You probably need to update your problem-specifications repository and re-gererate the tests.

**Comment by Insti** [comment]
I had a quick look, and this all looks good 👍 
Just the test data that needs updating to the latest version.

**Comment by jgr** [comment]
Thanks @Insti! I went and and pushed an update with updated tests.

**Comment by kotp** [comment]
Thank you @jgr!

**Pull request merged**